docs/using-oci-flexvolume-driver.md

```diff
@@ -0,0 +1,140 @@
+| `OCI_FLEXD_CONFIG_DIRECTORY` | `/usr/libexec/kubernetes/kubelet-plugins/volume/exec/oracle~oci` | Directory where the driver configuration lives |
+| `OCI_FLEXD_KUBECONFIG_PATH` | `/usr/libexec/kubernetes/kubelet-plugins/volume/exec/oracle~oci/kubeconfig` | An override to allow the fully qualified path of the kubeconfig resource file to be specified. This take precedence over additional configuration. |
+
+## OCI Policies
+
+You must ensure the user (or group) associated with the OCI credentials provided has the following level of access. See [policies][7] for more information.
+
+TODO: The below is now out of date as we require mounting in a kubeconfig.
+
+```
+"Allow group id GROUP to read vnic-attachments in compartment id COMPARTMENT",
+"Allow group id GROUP to read vnics in compartment id COMPARTMENT"
+"Allow group id GROUP to read instances in compartment id COMPARTMENT"
+"Allow group id GROUP to read subnets in compartment id COMPARTMENT"
+"Allow group id GROUP to use volumes in compartment id COMPARTMENT"
+"Allow group id GROUP to use instances in compartment id COMPARTMENT"
+"Allow group id GROUP to manage volume-attachments in compartment id COMPARTMENT"
+```
+
+### Kubernetes DaemonSet Installer
+
+The recommended way to install the driver is through the DaemonSet installer
+mechanism. This will create two DaemonSets, one specifically for master nodes,
+allowing configuration via a Kubernetes Secret, and one for worker nodes.
+
+```bash
+$ kubectl apply -f https://github.com/oracle/oci-flexvolume-driver/releases/download/${flexvolume_driver_version}/rbac.yaml
+$ kubectl apply -f https://github.com/oracle/oci-flexvolume-driver/releases/download/${flexvolume_driver_version}/oci-flexvolume-driver.yaml
+```
+
+You'll still need to add the config file as a Kubernetes Secret.
+
+Once the Secret is set and the daemonsets deployed, the configuration file will be placed onto the master nodes.
+
+## Assumptions
+
+- If a Flexvolume is specified for a Pod, it will only work with a single
+  replica. (or if there is more than one replica for a Pod, they will all have
+  to run on the same Kubernetes Node). This is because a volume can only be
+  attached to one instance at any one time. Note: This is in common with both
+  the Amazon and Google persistent volume implementations, which also have the
+  same constraint.
+
+- If nodes in the cluster span availability domain you must make sure your Pods are scheduled
+  in the correct availability domain. This can be achieved using the label selectors with the zone/region.
+
+  Using the [oci-volume-provisioner][3] makes this much easier.
+
+- For all nodes in the cluster, the instance display name in the OCI API must
+  match with the instance hostname, start with the vnic hostnamelabel or match the public IP.
+  This relies on the requirement that the nodename must be resolvable.
```

Comment:
This relies on the requirement that the nodename must be resolvable.
->
This is because it is required that all nodenames must be DNS resolvable.

Better ???